### PR TITLE
docs python classes

### PR DESCRIPTION
**Pull request by stefanozampini**
Just the matrix for now. @dalcinl @dham please review before opening the big merge request

### CODE REVIEW

**Review by dalcinl — APPROVED** [review]

File: src/binding/petsc4py/docs/source/petsc_python_types.rst [hunk 1]
```diff
@@ -1,8 +1,33 @@
...
+
+Here we discuss details about Python-aware PETSc types that can be used within the library.
+
 .. _petsc_python_mat:
 
 PETSc Python matrix type
 ========================
 
+PETSc provides a convenient way to compute the action of linear operators coded
+in Python through the MATPYTHON class.
```
Comment (dalcinl) [inline src/binding/petsc4py/docs/source/petsc_python_types.rst:14]: Not sure if this patch will actually work
```suggestion
in Python through the `~petsc4py.PETSc.Mat.Type.PYTHON` `Mat` subtype.
```

**Comment by stefanozampini** [comment]
just realized it is missing a few methods, view and setfromoptions. Will do

File: src/binding/petsc4py/docs/source/petsc_python_types.rst (hunk 1 above)
Comment (stefanozampini) [reply src/binding/petsc4py/docs/source/petsc_python_types.rst:14]: /Users/szampini/Devel/petsc_petsc4py_sprint/src/binding/petsc4py/docs/source/petsc_python_types.rst:13: WARNING: 'any' reference target not found: ~petsc4py.PETSc.Mat.Type.PYTHON

**Comment by stefanozampini** [comment]
PC is in too. We can leave KSP,SNES and TAO for after the big merge. @dham, @dalcinl please review

**Review by dham — APPROVED** [review]

**Pull request merged**